test/integration/test_language_container_deployer.py

```diff
@@ -0,0 +1,113 @@
+from typing import Callable
+from contextlib import ExitStack
+from pathlib import Path
+
+import pytest
+from _pytest.fixtures import FixtureRequest
+
+from pyexasol import ExaConnection
+from pytest_itde import config
+from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
+
+from exasol.python_extension_common.deployment.language_container_deployer import (
+    LanguageContainerDeployer, LanguageActivationLevel)
+
+from test.utils.revert_language_settings import revert_language_settings
+from test.utils.db_utils import (create_schema, assert_udf_running)
+
+
+def create_container_deployer(language_alias: str,
+                              pyexasol_connection: ExaConnection,
+                              bucketfs_config: config.BucketFs) -> LanguageContainerDeployer:
+    bucket_fs_factory = BucketFSFactory()
+    bucketfs_location = bucket_fs_factory.create_bucketfs_location(
+        url=f"{bucketfs_config.url}/default/container;bfsdefault",
+        user=f"{bucketfs_config.username}",
+        pwd=f"{bucketfs_config.password}",
+        base_path=None)
+    return LanguageContainerDeployer(
+        pyexasol_connection, language_alias, bucketfs_location)
+
+
+def test_language_container_deployer(
+        request: FixtureRequest,
+        connection_factory: Callable[[config.Exasol], ExaConnection],
+        exasol_config: config.Exasol,
+        bucketfs_config: config.BucketFs,
+        container_path: str):
+    """
+    Tests the deployment  of a container in one call, including the activation at the System level.
+    """
+    test_name: str = request.node.name
+    schema = test_name
+    language_alias = f"PYTHON3_TE_{test_name.upper()}"
```

Comment:
```suggestion
    language_alias = f"PYTHON3_PEC_{test_name.upper()}"
```

Reply:
* test_name.upper() might need sanitization should these tests every get parameterized 
* maybe extract into function, because you need to build the language_alias in many more tests

Reply:
Actually, I don't think we need to do anything fancy here. We always revert the language settings. So just a constant name, like "PYTHON3_PEC", would do. Same for the schema. We only need the schema to run a test udf. We don't need to create many of them.
What concerns me is that we use the same name for the container in the bucketfs in multiple tests. If some test fails to upload the container the test, theoretically, may still pass because the container is there from the previous test. On the other hand, making an uncontrolled number of copies of the container is not a good option either. Ideally, we would delete the container at the end of each test, but I am not sure how to delete an archive from the bucketfs.

Reply:
You are right, we reset the language settings, so it shouldn't be an issue to use a constant.

Regarding removing a file from bucketfs, in theory the bucketfs-python has functions for that, but I am not sure, if the API we are using has it. Furthermore, is removing a file from bucketfs a bit delicate, you might need to wait a bit before you can upload the a file under the same name.

Reply:
Maybe it is the best to wait for the container removal until we introduce the new bucketfs API, because it for sure supports removal. What do you think?

Reply:
Yes, the removal is finicky.  I've been experimenting with normal files recently. If you delete a file and try to upload another file with the same name immediately after (next test), it fails. I had to wait like 10-20 seconds. I don't know if we can delete an archive. I haven't tried that. Will it delete both the tar and the unzipped directory? How long will we have to wait after that?

Reply:
Yes, IT will removed both. Regarding the time or needs, that is a really good question. Probably, something we need to try and what also can be very different for SaaS.